Promact.Oauth.Server/src/Promact.Oauth.Server/Services/MessageServices.cs

```diff
@@ -40,7 +43,7 @@ public void SendEmail(string email, string subject, string message)
             using (var smtp = new SmtpClient())
             {
                 _logger.LogInformation("Smtp Connect");
-                smtp.Connect(_emailCrednetials.Value.From, _emailCrednetials.Value.Port, _emailCrednetials.Value.SslOnConnect == true ? MailKit.Security.SecureSocketOptions.SslOnConnect : MailKit.Security.SecureSocketOptions.None);
+                smtp.Connect(_emailCrednetials.Value.Host, _emailCrednetials.Value.Port, _emailCrednetials.Value.SetSmtpProtocol.ToLower() == _stringConstant.SetSmtpUnSecure ? MailKit.Security.SecureSocketOptions.None : _emailCrednetials.Value.SetSmtpProtocol.ToLower() == _stringConstant.SetSmtpSSL ? MailKit.Security.SecureSocketOptions.SslOnConnect : MailKit.Security.SecureSocketOptions.StartTls);
```

Comment:
This is complicated code. Refactor your logic into a separate method to make it more readable.